### ISSUE TITLE
Compilation error 22-x-y on MacOS 13.1: sprintf deprecated

### ISSUE THREAD

**Issue by mccare** [electron/build-tools#445]
I'm trying to compile electron 22-x-y branch on mac OS 13.1 ventura with downloaded xcode 14.1.0. 
and I get errors regarding the use of sprintf and the compile stops. I've managed to compile v23.0.0-alpha.3. 
Where would I disable the compiler flag to not break off compilation but just print out a warning and continue with the compliation?

Attached the log:

```
[1980/43695] CXX obj/third_party/harfbuzz-ng/harfbuzz_source/hb-subset.o
FAILED: obj/third_party/harfbuzz-ng/harfbuzz_source/hb-subset.o
/Users/chris/.electron_build_tools/third_party/goma/gomacc ../../third_party/llvm-build/Release+Asserts/bin/clang++ -MMD -MF obj/third_party/harfbuzz-ng/harfbuzz_source/hb-subset.o.d -DHAVE_OT -DHAVE_FREETYPE -DHAVE_ICU -DHAVE_ICU_BUILTIN -DHB_NO_MMAP -DHB_NO_RESOURCE_FORK -DHB_NO_SUBSET_LAYOUT -DHB_NO_FALLBACK_SHAPE -DHB_NO_UCD -DHB_NO_WIN1256 -DU_DISABLE_VERSION_SUFFIX=0 -DHB_NO_BUFFER_VERIFY -DHB_NO_DRAW -DHB_NO_PRAGMA_GCC_DIAGNOSTIC_ERROR -DHB_NO_PRAGMA_GCC_DIAGNOSTIC_WARNING -DDCHECK_ALWAYS_ON=1 -DCR_XCODE_VERSION=1410 -DCR_CLANG_REVISION=\"llvmorg-16-init-6578-g0d30e92f-2\" -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCXXABI_DISABLE_VISIBILITY_ANNOTATIONS -DCR_LIBCXX_REVISION=64d36e572d3f9719c5d75011a718f33f11126851 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DFT_CONFIG_MODULES_H=\"freetype-custom/freetype/config/ftmodule.h\" -DFT_CONFIG_OPTIONS_H=\"freetype-custom/freetype/config/ftoption.h\" -DPDFIUM_REQUIRED_MODULES -DCHROMIUM_RESTRICT_VISIBILITY -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -I../.. -Igen -I../../buildtools/third_party/libc++ -I../../third_party/freetype/include -I../../third_party/freetype/include/freetype-custom -I../../third_party/freetype/src/include -I../../third_party/harfbuzz-ng/src/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing -fstack-protector -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -ffp-contract=off -fcomplete-member-pointers -arch arm64 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -ffile-compilation-dir=. -no-canonical-prefixes -ftrivial-auto-var-init=pattern -fno-omit-frame-pointer -gdwarf-4 -g1 -gdwarf-aranges -isysroot sdk/xcode_links/MacOSX13.sdk -mmacos-version-min=10.13 -fvisibility=hidden -Xclang -add-plugin -Xclang find-bad-constructs -Xclang -plugin-arg-find-bad-constructs -Xclang raw-ref-template-as-trivial-member -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Werror -Wall -Wno-unused-variable -Wno-c++11-narrowing -Wno-unused-but-set-variable -Wno-misleading-indentation -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-psabi -Wloop-analysis -Wno-unneeded-internal-declaration -Wenum-compare-conditional -Wno-ignored-pragma-optimize -Wno-deprecated-builtins -Wno-bitfield-constant-conversion -O3 -std=c++20 -Wno-trigraphs -fno-exceptions -fno-rtti -nostdinc++ -isystem../../buildtools/third_party/libc++/trunk/include -isystem../../buildtools/third_party/libc++abi/trunk/include -fvisibility-inlines-hidden -c ../../third_party/harfbuzz-ng/src/src/hb-subset.cc -o obj/third_party/harfbuzz-ng/harfbuzz_source/hb-subset.o
In file included from ../../third_party/harfbuzz-ng/src/src/hb-subset.cc:51:
In file included from ../../third_party/harfbuzz-ng/src/src/hb-ot-layout-gsub-table.hh:32:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/GSUB.hh:6:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/SubstLookup.hh:5:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/SubstLookupSubTable.hh:6:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/MultipleSubst.hh:5:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/MultipleSubstFormat1.hh:5:
../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/Sequence.hh:120:2: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
        sprintf (p, "%u", i);
        ^
sdk/xcode_links/MacOSX13.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
sdk/xcode_links/MacOSX13.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
In file included from ../../third_party/harfbuzz-ng/src/src/hb-subset.cc:51:
In file included from ../../third_party/harfbuzz-ng/src/src/hb-ot-layout-gsub-table.hh:32:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/GSUB.hh:6:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/SubstLookup.hh:5:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/SubstLookupSubTable.hh:8:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/LigatureSubst.hh:5:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/LigatureSubstFormat1.hh:5:
In file included from ../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/LigatureSet.hh:5:
../../third_party/harfbuzz-ng/src/src/OT/Layout/GSUB/Ligature.hh:121:2: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
        sprintf (p, "%u", match_positions[i]);
        ^
sdk/xcode_links/MacOSX13.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
sdk/xcode_links/MacOSX13.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
                                                      ^
2 errors generated.
[2147/43694] CXX obj/third_party/icu/icui18n/rematch.o
ninja: build stopped: subcommand failed.
ERROR Error: Command failed: ninja -j 200 electron
    at checkExecSyncError (node:child_process:861:11)
    at Object.execFileSync (node:child_process:896:15)
    at Object.depotExecFileSync [as execFileSync] (/Users/chris/.electron_build_tools/src/utils/depot-tools.js:127:23)
    at runNinja (/Users/chris/.electron_build_tools/src/e-build.js:60:9)
    at Command.<anonymous> (/Users/chris/.electron_build_tools/src/e-build.js:121:7)
    at Command.listener [as _actionHandler] (/Users/chris/.electron_build_tools/node_modules/commander/lib/command.js:480:17)
    at /Users/chris/.electron_build_tools/node_modules/commander/lib/command.js:1234:65
    at Command._chainOrCall (/Users/chris/.electron_build_tools/node_modules/commander/lib/command.js:1151:12)
    at Command._parseCommand (/Users/chris/.electron_build_tools/node_modules/commander/lib/command.js:1234:27)
 ~  WorkArea  …  release  ERROR  % 
 ~  WorkArea  …  release  ERROR  % 
```

**Comment by stefansundin** [comment]
It looks like harfbuzz fixed this in the latest version, 6.0.0. https://github.com/harfbuzz/harfbuzz/releases/tag/6.0.0

chromium is using 5.3.1. https://chromium.googlesource.com/chromium/src/third_party/+/refs/heads/main/harfbuzz-ng/README.chromium

I was able to work around it with:

```
cd ~/electron/src/third_party/harfbuzz-ng/src
git remote add github https://github.com/harfbuzz/harfbuzz.git
git fetch github
git checkout 6.0.0
```

Not sure if there's a better way to do it. 🤷 

My build is still going on but it looks promising so far.

**Comment by mccare** [comment]
Since compilation with v23 works fine, feel free to close it.
